frontend/components/puzzle1.jsx

```diff
@@ -20,13 +27,19 @@ class Puzzle1 extends React.Component {
     var userinput = this.state.value;
     this.props.send('submit', { puzzle: '1', answer: userinput });
   }
+
+  cipherPing() {
+    this.props.send('cipher_ping', {});
+  }
   render() {
     const { value } = this.state;
     const { correct } = this.props;
+    const { cipher } = this.state
```

Comment:
multiline deconstructing is possible too: `const { value, cipher } = this.state;`